### PR TITLE
Modernize Setup

### PR DESCRIPTION
**Pull request by nlordell**
This PR modernizes the setup a bit of this repository:

- Switches over to using PNPM (nice experiences from `safe-modules` repository)
- Updates package version dependencies to the latest (with the exception of `eslint` which has an older version because of `typescript-eslint` peer dependency - i.e. TypeScript ESLint support is not compatible with ESLint v9)
- ESLint now uses the new ["flat configuration format"](https://eslint.org/docs/latest/use/configure/migration-guide) instead of the legacy format.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,2 @@
-build/
-node_modules/
-.DS_Store
```
Comment (remedcu) [inline .gitignore:3]: Non-blocking: Should we keep the `.DS_Store` so it is not accidentally merged when working in MacOS?

**Review by remedcu — APPROVED** [review]
Small nit, rest LGTM 👍🏾

File: .gitignore (hunk 1 above)
Comment (nlordell) [reply .gitignore:3]: Nice catch! Will ignore it. Been in Linux-land too long :stuck_out_tongue:

Comment (nlordell) [reply .gitignore:3]: Fixed in 445d20b

Comment (mmv08) [reply .gitignore:3]: An alternative POV shared with me when I contributed to the uniswap repo: `.DS_Store` is an OS-wide thing not unique to this project. Therefore, they require it to be a part of the global `.gitignore` file on their developer machines to not pollute the project's gitignore.

**Pull request merged**